crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/DeadLetterMessageServiceTest.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.laa.crime.crowncourt.service;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.laa.crime.crowncourt.entity.DeadLetterMessageEntity;
+import uk.gov.justice.laa.crime.crowncourt.prosecution_concluded.validator.ProsecutionConcludedValidator;
+import uk.gov.justice.laa.crime.crowncourt.repository.DeadLetterMessageRepository;
+
+@ExtendWith(MockitoExtension.class)
+class DeadLetterMessageServiceTest {
+    @InjectMocks
+    private DeadLetterMessageService deadLetterMessageService;
+
+    @Spy
+    private DeadLetterMessageRepository deadLetterMessageRepository;
+
+    @Captor
+    private ArgumentCaptor<DeadLetterMessageEntity> deadLetterMessageCaptor;
+
+    @Test
+    void givenErrorReasonAndQueueMessage_whenLogDeadLetterMessageIsInvoked_thenDeadLetterMessageIsCreated() {
+        String errorReason = ProsecutionConcludedValidator.MAAT_ID_FORMAT_INCORRECT;
+        String queueMessage = "{ \"maatId\": \"A-123456\" }";
+
+        deadLetterMessageService.logDeadLetterMessage(errorReason, queueMessage);
+
+        verify(deadLetterMessageRepository, times(1)).save(deadLetterMessageCaptor.capture());
+        DeadLetterMessageEntity deadLetterMessageEntity = deadLetterMessageCaptor.getValue();
+
+        Assertions.assertEquals(errorReason, deadLetterMessageEntity.getDeadLetterReason());
+        Assertions.assertNotNull(deadLetterMessageEntity.getMessage());
```

Comment:
We use `assertJ` in our tests, whilst these are JUnit5 assertions, we should update these to maintain consistency.

Reply:
Done